src/AppBundle/Controller/CoDeputyController.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace AppBundle\Controller;
+
+use AppBundle\Entity as EntityDir;
+use JMS\Serializer\Exception\RuntimeException;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Symfony\Component\HttpFoundation\Request;
+
+
+/**
+ * @Route("/codeputy/")
+ */
+class CoDeputyController extends RestController
+{
+
+    /**
+     * @Route("add")
+     * @Method({"POST"})
+     */
+    public function add(Request $request)
+    {
+        $this->denyAccessUnlessGranted([EntityDir\User::ROLE_DEPUTY]);
+
+        $data = $this->deserializeBodyContent($request, [
+            'email' => 'notEmpty',
+        ]);
+
+        $loggedInUser = $this->getUser();
+        $newUser = new EntityDir\User();
+
+        $newUser->setFirstname('');
+        $newUser->setLastname('');
```

Comment:
wont a new User object have an empty first and last name? Not sure above 2 statements are required